vaadin-combo-box-flow-parent/vaadin-combo-box-flow/src/main/java/com/vaadin/flow/component/combobox/ComboBoxBase.java

```diff
@@ -410,9 +478,27 @@ public void setRenderer(Renderer<TItem> renderer) {
         renderManager.setRenderer(renderer);
     }
 
+    @Override
+    public void setValue(TValue value) {
+        if (getDataCommunicator() == null
+                || getDataProvider() instanceof DataCommunicator.EmptyDataProvider) {
+            if (value == getEmptyValue()) {
```

Comment:
Changed to use `getEmptyValue()` instead of `null`. That should be equivalent for `ComboBox`. For `MultiSelectComboBox` we need should deal with `null` separately in an override - either throw as illegal argument, or convert to empty value.